### PR TITLE
REM-563

### PR DESCRIPTION
**Pull request by RebeccaFrem02**
add HR filters to RPB list
you can test on Anthonys or neg-deploy

### CODE REVIEW

**Comment by imtithalhalalo** [comment]
Screenshot 2024-10-02 132228
not showing the name for Employee Ref

File: src/resources/ResourceIds.js [hunk 1]
```diff
@@ -248,6 +248,22 @@ export const ResourceIds = {
   AgingButton: 31419,
   LotCategories: 41108,
   ClientBalance: 36131,
+  SalesDefaults: 51114,
```
Comment (imtithalhalalo) [inline src/resources/ResourceIds.js:251]: already exists

**Comment by imtithalhalalo** [comment]
Screenshot 2024-10-02 132649
it is showing this error in Turnover Rate page

File: src/components/Shared/apiMappings.js [hunk 2]
```diff
@@ -713,5 +710,143 @@ export const apiMappings = {
...
+    parameters: `_filter=&_size=1000&_startAt=0&_type=0&_activeStatus=0&_sortBy=recordId`,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
+    columnsInDropDown: [{ key: 'name', value: 'Name' }]
+  },
+  [ResourceIds.CertificateFilter]: {
+    type: COMBOBOX,
+    endpoint: EmployeeRepository.CertificateFilters.qry,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
```
Comment (imtithalhalalo) [inline src/components/Shared/apiMappings.js:775]: here you added display reference and name but in the columnsInDropdown only name is that how it should be?

File: src/components/Shared/apiMappings.js [hunk 3]
```diff
@@ -713,5 +710,143 @@ export const apiMappings = {
...
+    columnsInDropDown: [
+      { key: 'reference', value: 'Reference' },
+      { key: 'name', value: 'Name' }
+    ]
+  },
+  [ResourceIds.SalaryChangeReasonFilter]: {
+    type: COMBOBOX,
+    endpoint: EmployeeRepository.SalaryChangeReasonFilters.qry,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
```
Comment (imtithalhalalo) [inline src/components/Shared/apiMappings.js:792]: also here

File: src/repositories/EmployeeRepository.js [hunk 4]
```diff
@@ -1,9 +1,25 @@
...
+    qry: service + 'qrySC'
+  },
+  EmploymentStatusFilters: {
+    qry: service + 'qryST'
+  },
+  SponsorFilters: {
+    qry: service + 'qrySP'
+  },
+  EmployeeFilters: {
+    snapshot: service + 'snapshotEM'
```
Comment (imtithalhalalo) [inline src/repositories/EmployeeRepository.js:23]: this exists on line 5

File: src/resources/ResourceIds.js (hunk 1 above)
Comment (RebeccaFrem02) [reply src/resources/ResourceIds.js:251]: where?

Comment (imtithalhalalo) [reply src/resources/ResourceIds.js:251]: at the end

File: src/components/Shared/apiMappings.js [hunk 5]
```diff
@@ -713,5 +710,143 @@ export const apiMappings = {
...
+    columnsInDropDown: [{ key: 'name', value: 'Name' }]
+  },
+  [ResourceIds.DepartmentFilter]: {
+    type: COMBOBOX,
+    endpoint: companyStructureRepository.DepartmentFilters.qry,
+    parameters: `_filter=&_size=1000&_startAt=0&_type=0&_activeStatus=0&_sortBy=recordId`,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
+    columnsInDropDown: [{ key: 'name', value: 'Name' }]
```
Comment (NoraBitarSM) [inline src/components/Shared/apiMappings.js:769]: show departmentRef and name in the dropdown and on selection please

File: src/components/Shared/apiMappings.js [hunk 6]
```diff
@@ -713,5 +710,143 @@ export const apiMappings = {
...
+      { key: 'name', value: 'Name' }
+    ]
+  },
+  [ResourceIds.CompanyPosition]: {
+    type: COMBOBOX,
+    endpoint: companyStructureRepository.CompanyPositions.qry,
+    parameters: `_filter=&_size=1000&_startAt=0&_sortBy=recordId`,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
+    columnsInDropDown: [{ key: 'name', value: 'Name' }]
```
Comment (NoraBitarSM) [inline src/components/Shared/apiMappings.js:761]: please show positionRef and name in the dropdown and on selection

File: src/components/Shared/apiMappings.js [hunk 7]
```diff
@@ -713,5 +710,143 @@ export const apiMappings = {
...
+    endpoint: SystemRepository.GovernmentOrganization.qry,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
+    columnsInDropDown: [{ key: 'name', value: 'Name' }]
+  },
+  [ResourceIds.DivisionFilter]: {
+    type: COMBOBOX,
+    endpoint: companyStructureRepository.DivisionFilters.qry,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
```
Comment (NoraBitarSM) [inline src/components/Shared/apiMappings.js:742]: division doesn't have a reference so you can remove it from the displayField

**Review by NoraBitarSM — CHANGES_REQUESTED** [review]
please check for the rest of the HR filters for the fields in dropdown and on selection so the code can stay consistent

**Pull request merged**